java/org/apache/coyote/http11/Http11Processor.java

```diff
@@ -999,7 +999,12 @@ protected final void prepareResponse() throws IOException {
                     int keepAliveTimeout = protocol.getKeepAliveTimeout();
 
                     if (keepAliveTimeout > 0) {
-                        String value = "timeout=" + keepAliveTimeout / 1000L;
+                        StringBuilder value = new StringBuilder();
+                        value.append("timeout=").append(keepAliveTimeout / 1000L);
+                        int maxKeepAliveRequests = protocol.getMaxKeepAliveRequests();
+                        if (maxKeepAliveRequests > 0) {
```

Comment:
How is this idea implemented here? Because it affects keepAliveLeft.

[AbstractEndpoint](https://github.com/apache/tomcat/blob/main/java/org/apache/tomcat/util/net/AbstractEndpoint.java)
```
public int getMaxKeepAliveRequests() {
    // Disable keep-alive if the server socket is not bound
    if (bindState.isBound()) {
        return maxKeepAliveRequests;
    } else {
        return 1;
    }
}
```

Reply:
> If maxKeepAliveRequests is `-1` then set `max` to Integer.MAX_VALUE

I haven't found any RFC documents about how to response this parameter if no limit on server.
I think personally that if we don't limit the number of request on the connection. We should ignore this parameter, and leave it up to the client to decide how to handle this connection (e.g. use their default strategy). In additional, that's how **httpd** handles it if I read correct.

https://github.com/apache/httpd/blob/d62d3143f42771a7111c2148837c0a91a271b83d/modules/http/http_protocol.c#L287-L299